job.c

```diff
@@ -276,6 +277,60 @@ sig_dfl(void)
     signal(SIGPIPE, SIG_DFL);
 }
 
+#define DQUOTE '\"'
+#define BSLASH '\\'
+#define SPECIALS "()<>[].,:;@"
+#define MAIL_LINE_LEN 998             /* RFC5322 */
+
+char *
+make_mailbox(char *displayname, char *mail_from, char *hostname)
+    /* Produce a "mailbox" header as per RFC5322 sec. 3.2.3
+     * <https://datatracker.ietf.org/doc/html/rfc5322#section-3.2.3> */
+{
+    char specials[sizeof(SPECIALS) + 2];
+    char need_quotes = 0, need_anglebrackets = 0;
+    char c = '\0';
+    char *bpos = NULL, *dpos = NULL;
+    char *buf1 = NULL, *buf2 = NULL;
+
+    snprintf(specials, sizeof(specials), "%s%c%c", SPECIALS, DQUOTE, BSLASH);
+
+    /* This should actually be shorter 'cause the header name is prepended
+     * downstream */
+    buf1 = (char *)calloc(MAIL_LINE_LEN+1, sizeof(char));
+    buf2 = (char *)calloc(MAIL_LINE_LEN+1, sizeof(char));
+
+    /* walk the displayname and rebuild it in buf1 */
+    dpos = displayname;
+    bpos = buf1;
+    while (*dpos) {
+        c = *dpos++;
+        if (strchr(specials, c)) {
+            /* insert escape */
+            if (c == DQUOTE) *bpos++ = BSLASH;
```

Comment:
`displayname` comes from `read_conf()` which has a buffer of 1024, which makes an upper bound over 1000 for `displayname`.
And here you add extra characters on top of that.

We are at risk of overflowing `buf1` which is of size 998: can you add checks on size?

Reply:
👍 
Now thinking about that, RFC recommends a soft max of 78. Should we add a warning maybe downstream (where the full mailbox line is put together: job.c:377) for that?